src/pkg/storage/storage.go

```diff
@@ -24,7 +25,7 @@ var (
 // Storage defines a storage provider, along with any configuration
 // required to set up or communicate with that provider.
 type Storage struct {
-	Provider storageProvider
+	Provider Provider
```

Comment:
Need to fix this. Converting diff to draft for now.

Reply:
Re: `Provider Provider` field.

Looks like naming a struct field the same name as its type is idiomatic as per golang conventions. Also, `Type` suffix is discouraged. @ryanfkeepers Need your thoughts on this! Should I keep this as is, i.e. `Provider Provider`?